preset/plugin/src/main/kotlin/ch/ubique/gradle/preset/PresetPlugin.kt

```diff
@@ -19,39 +21,12 @@ abstract class PresetPlugin : Plugin<Project> {
 		val androidExtension = getAndroidExtension(project)
 		val androidComponentExtension = getAndroidComponentsExtension(project)
 
+		// Apply presets specific to applications
+		(androidExtension as? ApplicationExtension)?.applyAppPreset(project)
+
 		// Enable BuildConfig
 		androidExtension.buildFeatures.buildConfig = true
 
-		// Default dimension
-		androidExtension.flavorDimensions("default")
-
-		// Add flavor boolean fields to BuildConfig
-		androidExtension.productFlavors.configureEach { flavor ->
-			val sanitizedFlavorName = flavor.name.replace("[^a-zA-Z0-9_]", "_")
-
-			// default flavor dimension
-			flavor.dimension = "default"
-
-			// default application id suffix
-			flavor.applicationIdSuffix = when (flavor.name) {
-				"prod", "production" -> null
-				else -> ".$sanitizedFlavorName"
-			}
-
-			// flavor BuildConfig flag
-			val flavorFieldName = "IS_FLAVOR_${sanitizedFlavorName.uppercase()}"
-			// true for this flavor ...
-			flavor.buildConfigField("boolean", flavorFieldName, "true")
-			// ... false for all others
-			androidExtension.defaultConfig.buildConfigField("boolean", flavorFieldName, "false")
-		}
-
-		// Release build config
-		androidExtension.buildTypes.maybeCreate("release").apply {
-			isMinifyEnabled = true
-			proguardFiles(getDefaultProguardFile("proguard-android.txt", project.layout.buildDirectory), "proguard-rules.pro")
-		}
-
 		// R8 full mode check
 		if (project.findProperty("android.enableR8.fullMode") != "false" && project.findProperty("android.enableR8.fullModeAllowed") != "true") {
 			throw IllegalArgumentException("R8 full mode is enabled. Disable it with android.enableR8.fullMode=false or allow it by setting android.enableR8.fullModeAllowed=true")
```

Comment:
Sollte der Check nicht auch in die App-only Config? Proguard ist ja eigentlich nur ein App-Thema.